plugins/pulp_rpm/plugins/importers/yum/upload.py

```diff
@@ -456,7 +456,11 @@ def _handle_package(repo, type_id, unit_key, metadata, file_path, conduit, confi
     # check if the unit has duplicate nevra
     purge.remove_unit_duplicate_nevra(unit, repo)
 
-    unit.set_storage_path(rpm_basefilename)
+    if type_id != models.DRPM._content_type_id.default:
+        unit.set_storage_path(rpm_basefilename)
+    else:
+        unit.set_storage_path(unit.filename)
```

Comment:
i think you can use unit.filename in both cases, since rpm_basefilename gets assigned to unit.filename on L452 wdyt?

Reply:
I did not even notice that - great catch! Lemme fix and repush